test/e2e/app_management_ns_test.go

```diff
@@ -1993,12 +2004,20 @@ func TestNamespacedNamespaceAutoCreationWithMetadataAndNsManifest(t *testing.T)
 		Then().
 		Expect(Success("")).
 		Expect(Namespace(namespace, func(app *Application, ns *v1.Namespace) {
+			assert.NotEmpty(t, app.Status.Conditions)
+
+			trackingId := ns.Annotations["argocd.argoproj.io/tracking-id"]
+
 			delete(ns.Labels, "kubernetes.io/metadata.name")
 			delete(ns.Labels, "argocd.argoproj.io/tracking-id")
 			delete(ns.Labels, "kubectl.kubernetes.io/last-applied-configuration")
 			delete(ns.Annotations, "argocd.argoproj.io/tracking-id")
 			delete(ns.Annotations, "kubectl.kubernetes.io/last-applied-configuration")
 
+			// TODO: The tracking id is different for namespace manifests vs namespaces generated with CreateNamespace=true
+			// Should that be the case or should we change the format to be the same for both types?
+			assert.Equal(t, fmt.Sprintf("%s:/Namespace:%s/%s", app.Name, AppNamespace(), namespace), trackingId)
```

Comment:
I'm not sure why we get a `argocd-e2e-external_` prefix when we have a namespace manifest